src/app/modules/share/header/header.component.ts

```diff
@@ -1,39 +1,101 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import {Component, EventEmitter, OnDestroy, OnInit, Output} from '@angular/core';
 import { Router, NavigationEnd } from "@angular/router"
-import { filter } from 'rxjs/operators';
-import { Subscription } from "rxjs";
+import { filter, takeUntil } from 'rxjs/operators';
+import { Subject, Subscription } from "rxjs";
+import { UserInfoService } from "./user-info.service";
+import { AuthService } from "../../../services/auth.service";
 
 @Component({
   selector: 'ls-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss']
 })
 export class HeaderComponent implements OnInit, OnDestroy {
-  private subscription: Subscription;
+  private navSubscription: Subscription;
+  private authSubscription: Subject<any> = new Subject();
+  private userInfoSubscription: Subject<any> = new Subject();
```

Comment:
U can use one subject for unsubscribe from subscriptions in couple with takeUntil operator.

For example: `destroy$ = new Subject();`

Reply:
Fixed